azext_iot/iothub/providers/state.py

```diff
@@ -100,27 +117,77 @@ def process_hub(self, target: dict):
 
             if id["capabilities"]["iotEdge"]:
                 children = _iot_device_children_list(target=target, device_id=id["deviceId"])
-                hub_state["children"][id["deviceId"]] = [c["deviceId"] for c in children]
+                hub_state["devices"]["children"][id["deviceId"]] = [c["deviceId"] for c in children]
 
-            hub_state["devices"][id["deviceId"]] = identities[i]
+            identities[i]["properties"].pop("reported")
+            [identities[i]["properties"]["desired"].pop(key) for key in ["$metadata", "$version"]]
+            [identities[i].pop(key, None) for key in ["cloudToDeviceMessageCount", "configurations", "deviceEtag", "deviceScope",
+                                                      "lastActivityTime", "modelId", "parentScopes", "statusUpdateTime", "etag",
+                                                      "version"]]
+            hub_state["devices"]["identities"].append(identities[i])
 
             for module in module_objs:
                 module = vars(module)
 
                 if module["module_id"] not in ["$edgeAgent", "$edgeHub"]:
 
-                    # these can't be json serialized, and they can't be explicitly set anyways
-                    module.pop('connection_state_updated_time')
-                    module.pop('last_activity_time')
-
                     module_twin = _iot_device_module_twin_show(target=target, device_id=id["deviceId"],
                                                                module_id=module["module_id"])
 
                     module2 = _iot_device_module_show(target=target, device_id=id["deviceId"], module_id=module["module_id"])
                     module["authentication"] = module2["authentication"]
 
+                    [module.pop(key) for key in ["connection_state_updated_time", "last_activity_time", "etag",
```

Comment:
The list of keys should be a constant, there are more things within this file that should be constants